### PR TITLE
Feat port test t1b

### PR DESCRIPTION
**Pull request by jaiszeen**
**Issue #, if available:**

**Description of changes:**
Port test Logmanager-T1-b
As a customer I can configure the logs uploader component using a componentLogsConfigurationMap and logs are uploaded to CloudWatch
**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] Updated or added new end-to-end tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/src/main/java/com.aws.greengrass/FileSteps.java [hunk 1]
```diff
@@ -140,6 +141,28 @@ public void verifyActiveFile(String componentName) {
...
         assertEquals(expectedActiveFilePath, activeFile.getAbsolutePath());
     }
+    /**
+     * Arranges some log files with content on the /logs folder for a component
+     * to simulate a devices where logs have already bee written.
+     *
+     * @param componentName name of the component.
+     */
+
+    @And("I verify the rotated files are not deleted except for the active log file for component {word}")
```
Comment (Nelsonochoam) [inline uat/src/main/java/com.aws.greengrass/FileSteps.java:151]: Delete this?

File: uat/src/main/resources/greengrass/features/log-manager-1.feature [hunk 2]
```diff
@@ -44,6 +44,39 @@ Feature: Greengrass V2 LogManager
...
+            | aws.greengrass.LogManager     | LATEST |
+        When I update my Greengrass deployment configuration, setting the component aws.greengrass.LogManager configuration to:
+            """
+            {
+                "MERGE": {
+                    "logsUploaderConfiguration": {
+                        "componentLogsConfigurationMap": {
+                             "UserComponentA": {
+                            "logFileRegex": "UserComponentA_\\w*.log",
+                            "logFileDirectoryPath": "${UserComponentALogDirectory}"
```
Comment (Nelsonochoam) [inline uat/src/main/resources/greengrass/features/log-manager-1.feature:59]: Indentation

File: uat/src/main/resources/greengrass/features/log-manager-1.feature [hunk 3]
```diff
@@ -44,6 +44,39 @@ Feature: Greengrass V2 LogManager
...
+                             }
+                        },
+                            "systemLogsConfiguration": {
```
Comment (Nelsonochoam) [inline uat/src/main/resources/greengrass/features/log-manager-1.feature:62]: Indentation

File: uat/src/main/java/com.aws.greengrass/FileSteps.java [hunk 4]
```diff
@@ -113,13 +114,12 @@ private void addDataToFile(String data, Path filePath) throws IOException {
...
         }
     }
-
     /**
      * Arranges some log files with content on the /logs folder for a component
      * to simulate a devices where logs have already bee written.
-     *
      * @param componentName name of the component.
      */
+
```
Comment (Nelsonochoam) [inline uat/src/main/java/com.aws.greengrass/FileSteps.java:122]: Extras line here

File: uat/src/main/resources/greengrass/features/log-manager-1.feature [hunk 5]
```diff
@@ -44,37 +44,71 @@ Feature: Greengrass V2 LogManager
...
+                            "logFileDirectoryPath": "${UserComponentALogDirectory}"
+                        }
+                    },
+                    "systemLogsConfiguration": {
+                        "uploadToCloudWatch": "true",
+                        "minimumLogLevel": "INFO",
+                        "diskSpaceLimit": "25",
+                        "diskSpaceLimitUnit": "MB",
+                        "deleteLogFileAfterCloudUpload": "true"
+                        }
```
Comment (Nelsonochoam) [inline uat/src/main/resources/greengrass/features/log-manager-1.feature:68]: Indentation

File: uat/src/main/resources/greengrass/features/log-manager-1.feature [hunk 6]
```diff
@@ -44,37 +44,71 @@ Feature: Greengrass V2 LogManager
...
                     },
-                    "periodicUploadIntervalSec": "10"
-                }
+                    "systemLogsConfiguration": {
+                        "uploadToCloudWatch": "true",
+                        "minimumLogLevel": "INFO",
+                        "diskSpaceLimit": "25",
+                        "diskSpaceLimitUnit": "MB",
+                        "deleteLogFileAfterCloudUpload": "true"
+                        }
```
Comment (Nelsonochoam) [inline uat/src/main/resources/greengrass/features/log-manager-1.feature:103]: Indentation

File: uat/src/main/java/com.aws.greengrass/FileSteps.java [hunk 7]
```diff
@@ -113,13 +114,12 @@ private void addDataToFile(String data, Path filePath) throws IOException {
             writer.write(data + "\r\n");
         }
     }
-
     /**
```
Comment (Nelsonochoam) [inline uat/src/main/java/com.aws.greengrass/FileSteps.java:117]: Add this space back checkstyle will complain. Make sure you compile before pushing

**Review by Nelsonochoam — APPROVED** [review]

**Review by saranyailla — APPROVED** [review]

**Pull request merged**